File: rea2/app/src/main/java/com/jang/assistales/model/FaunaBean.java
package com.jang.assistales.model;

import java.io.Serializable;

public class FaunaBean implements Serializable {
    private String name;
    private String description;
    private String race;
    private String aggressive;
    private String attack;
    private String defence;
    private String intelligence;
    private String ability;
    private String agility;
    private boolean group;
    private String power;
    private String weakness;
    private boolean rarity;
    private String ear;
    private String muzzle;
    private String tail;
    private String fur;
    private String color;
    private String particularity;

    @Override
    public String toString() {
        return name;
    }

    //--------------------------Constructor---------------------------//

    public FaunaBean(String name) {
        this.name = name;
    }


    //--------------------------Getter/Setter-------------------------//


    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getRace() {
        return race;
    }

    public void setRace(String race) {
        this.race = race;
    }

    public String getAggressive() {
        return aggressive;
    }

    public void setAggressive(String aggressive) {
        this.aggressive = aggressive;
    }

    public String getAttack() {
        return attack;
    }

    public void setAttack(String attack) {
        this.attack = attack;
    }

    public String getDefence() {
        return defence;
    }

    public void setDefence(String defence) {
        this.defence = defence;
    }

    public String getIntelligence() {
        return intelligence;
    }

    public void setIntelligence(String intelligence) {
        this.intelligence = intelligence;
    }

    public String getAbility() {
        return ability;
    }

    public void setAbility(String ability) {
        this.ability = ability;
    }

    public String getAgility() {
        return agility;
    }

    public void setAgility(String agility) {
        this.agility = agility;
    }

    public boolean isGroup() {
        return group;
    }

    public void setGroup(boolean group) {
        this.group = group;
    }

    public String getPower() {
        return power;
    }

    public void setPower(String power) {
        this.power = power;
    }

    public String getWeakness() {
        return weakness;
    }

    public void setWeakness(String weakness) {
        this.weakness = weakness;
    }

    public boolean isRarity() {
        return rarity;
    }

    public void setRarity(boolean rarity) {
        this.rarity = rarity;
    }

    public String getEar() {
        return ear;
    }

    public void setEar(String ear) {
        this.ear = ear;
    }

    public String getMuzzle() {
        return muzzle;
    }

    public void setMuzzle(String muzzle) {
        this.muzzle = muzzle;
    }

    public String getTail() {
        return tail;
    }

    public void setTail(String tail) {
        this.tail = tail;
    }

    public String getFur() {
        return fur;
    }

    public void setFur(String fur) {
        this.fur = fur;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getParticularity() {
        return particularity;
    }

    public void setParticularity(String particularity) {
        this.particularity = particularity;
    }
}

File: rea5/assistales/app/src/main/java/com/jang/assistales/model/FaunaBean.kt
package com.jang.assistales.model

import androidx.room.*

@Entity(
    tableName = "faune",
    foreignKeys = [ForeignKey(
        entity = UniversBean::class,
        parentColumns = ["id_univers"],
        childColumns = ["universId"]
    )]
)
class FaunaBean(
    val name: String,
    @PrimaryKey(autoGenerate = true)
    val id_fauna: Long = 0,
    val description: String? = null,
    val race: String? = null,
    val aggressive: String? = null,
    val attack: String? = null,
    val defence: String? = null,
    val intelligence: String? = null,
    val ability: String? = null,
    val agility: String? = null,
    val group: Boolean = false,
    val power: String? = null,
    val weakness: String? = null,
    val rarity: Boolean = false,
    val ear: String? = null,
    val muzzle: String? = null,
    val tail: String? = null,
    val fur: String? = null,
    val color: String? = null,
    val particularity: String? = null,
    val universId: Long = 0

) {
    override fun toString(): String {
        return name
    }


    @Dao
    interface FaunaDao {
        @Query("SELECT * FROM faune ORDER BY name ASC")
        fun getAll(): List<FaunaBean>

        @Insert
        fun insert(fauna: FaunaBean): Long

        @Update
        fun update(fauna: FaunaBean)

        @Delete
        fun delete(fauna: FaunaBean)
    }

}